mutt/date.c

```diff
@@ -699,3 +694,52 @@ time_t mutt_date_add_timeout(time_t now, long timeout)
 
   return now + timeout;
 }
+
+/**
+ * mutt_date_localtime - Converts timep to struct tm expressed in user timezone.
+ * @param timep Time
+ * @retval tm   Broken-down time representation
+ *
+ * Uses current time if timep is #MUTT_DATE_NOW
+ */
+struct tm mutt_date_localtime(time_t timep)
+{
+  struct tm tm = { 0 };
+
+  if (timep == MUTT_DATE_NOW)
+    timep = time(NULL);
+
+  localtime_r(&timep, &tm);
+  return tm;
+}
+
+/**
+ * mutt_date_gmtime - Converts timep to struct tm expressed in UTC timezone.
+ * @param timep Time
+ * @retval tm   Broken-down time representation
+ *
+ * Uses current time if timep is #MUTT_DATE_NOW
+ */
+struct tm mutt_date_gmtime(time_t timep)
+{
+  struct tm tm = { 0 };
+
+  if (timep == MUTT_DATE_NOW)
+    timep = time(NULL);
+
+  gmtime_r(&timep, &tm);
+  return tm;
+}
+
+/**
+ * mutt_date_localtime_format - Format localtime
+ * @param time   Time to format
+ * @param str    String to store formated time
+ * @param size   String size
+ * @param format Format to apply
+ */
+void mutt_date_localtime_format(time_t time, char *str, size_t size, char *format)
+{
+  struct tm tm = mutt_date_localtime(time);
+  strftime(str, size, format, &tm);
```

Comment:
`strftime()` returns the number of characters written to the buffer.
It might be useful to return that to the caller (for error checking).